test/integration/utils.go

```diff
@@ -55,3 +64,28 @@ func deleteAllEtcdKeys() {
 	}
 
 }
+
+func runAMaster(t *testing.T) (*master.Master, *httptest.Server) {
+	helper, err := master.NewEtcdHelper(newEtcdClient(), testapi.Version())
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	var m *master.Master
```

Comment:
I know you just moved this code, but there's no reason to do it in this order. Create the master first, then create the http server.